src/mixxx.cpp

```diff
@@ -1510,6 +1515,27 @@ void MixxxMainWindow::initActions()
     }
 }
 
+void MixxxMainWindow::slotUpdateWindowTitle(TrackPointer pTrack)
+{
+    QString title;
+    if (pTrack) {
+        QString trackTitle = pTrack->getTitle();
+        QString trackArtist = pTrack->getArtist();
+
+        //Displaying title as "Track Title" by "Artist" - "Current Version"
+        //Or "Track Title - "Current Version" in case the artist field is empty
+        if (trackArtist.size() != 0) {
+            title = trackArtist + " - " + trackTitle;
+        } else {
+            title = trackTitle;
+        }
+        title = title + " - " + m_MixxxVersion;
```

Comment:
Just noticed the GitHub title bar above and it looks somehow smooth.  
"$artist · $title - Mixxx"

On the other hand "$title by $artist - Mixxx" already works for me and has the advantage that we see the $titel... in a truncated taskbar button. 
This was @jercaianu first solution, he did already a cycle and I am afraid he likes to kill us for and extra cycles driven by such taste discussion.